packages/yum.go

```diff
@@ -180,6 +212,17 @@ func listAndParseYumPackages(ctx context.Context, opts ...YumUpdateOption) ([]*P
 		return nil, nil
 	}
 
+	// Some versions of yum will leave a transaction file in /tmp when update
+	// is run with --assumeno. This will attempt to delete that file.
+	yumTXFile := getYumTXFile(stdout)
```

Comment:
I'm wondering, if yum fails won't `ptyrunner.Run()` return an error? Also, even if yum doesn't fail it's very likely that yum will produce stdout and we're checking for stdout before the execution reaches this block and returning. What am I missing?

Reply:
on 211 we return `if stdout == nil`

Reply:
Yes, I don't think this is an issue. If there isn't anything in stdout, it will exit above. If there is anything in stdout, then this function call will be safe to call no matter what's in stdout.